backend/pkg/api/data_access/vdb_blocks.go

```diff
@@ -32,319 +41,272 @@ func (d *DataAccessService) GetValidatorDashboardBlocks(ctx context.Context, das
-	LEFT JOIN LATERAL (SELECT exec_block_hash, proposer_fee_recipient, max(value) as value
-        FROM relays_blocks
-        WHERE relays_blocks.exec_block_hash = blocks.exec_block_hash
-		GROUP BY exec_block_hash, proposer_fee_recipient
-    ) rb ON rb.exec_block_hash = blocks.exec_block_hash
-	)
-	`
-
-	from := `past_blocks `
-	selectStr := `SELECT * FROM `
+	// -------------------------------------
+	// Execute query
+	var proposals []struct {
+		Proposer     t.VDBValidator      `db:"validator_index"`
+		Group        uint64              `db:"group_id"`
+		Epoch        uint64              `db:"epoch"`
+		Slot         uint64              `db:"slot"`
+		Status       uint64              `db:"status"`
+		Block        sql.NullInt64       `db:"exec_block_number"`
+		FeeRecipient []byte              `db:"fee_recipient"`
+		ElReward     decimal.NullDecimal `db:"el_reward"`
+		ClReward     decimal.NullDecimal `db:"cl_reward"`
+		GraffitiText sql.NullString      `db:"graffiti_text"`
 
-	query := selectStr + from + where + orderBy + limitStr
-	// supply scheduled proposals, if any
-	if len(scheduledProposers) > 0 {
-		// distinct to filter out duplicates in an edge case (if dutiesInfo didn't update yet after a block was proposed, but the blocks table was)
-		// might be possible to remove this once the TODO in service_slot_viz.go:startSlotVizDataService is resolved
-		params = append(params, scheduledProposers)
-		params = append(params, scheduledEpochs)
-		params = append(params, scheduledSlots)
-		cte += fmt.Sprintf(`,
-		scheduled_blocks as (
-			SELECT
-			prov.proposer,
-			prov.epoch,
-			prov.slot,
-			%s,
-			'0'::text AS status,
-			NULL::int AS exec_block_number,
-			''::bytea AS fee_recipient,
-			NULL::float AS el_reward,
-			NULL::float AS cl_reward,
-			''::text AS graffiti_text
-		FROM unnest($%d::int[], $%d::int[], $%d::int[]) AS prov(proposer, epoch, slot)
-		`, groupIdCol, len(params)-2, len(params)-1, len(params))
-		if dashboardId.Validators == nil {
-			// add group id
-			cte += fmt.Sprintf(`INNER JOIN users_val_dashboards_validators validators 
-			ON validators.dashboard_id = $1 
-			AND validators.validator_index = ANY($%d::int[])
-			`, len(params)-2)
-		}
-		cte += `) `
-		distinct := "slot"
-		if !onlyPrimarySort {
-			distinct = sortColName + ", " + distinct
-		}
-		// keep all ordering, sorting etc
-		selectStr = `SELECT DISTINCT ON (` + distinct + `) * FROM `
-		// encapsulate past blocks query to ensure performance
-		from = `(
-			( ` + query + ` )
-			UNION ALL
-			SELECT * FROM scheduled_blocks
-		) as combined
-		`
-		// make sure the distinct clause filters out the correct duplicated row (e.g. block=nil)
-		orderBy += `, exec_block_number NULLS LAST`
-		query = selectStr + from + where + orderBy + limitStr
+		// for cursor only
+		Reward decimal.Decimal
 	}
-
 	startTime := time.Now()
-	err = d.alloyReader.SelectContext(ctx, &proposals, cte+query, params...)
+	query, args, err := blocksDs.Prepared(true).ToSQL()
```

Comment:
**issue:** Guest dashboards do not work, you get an error here
`"unsupported type dataaccess.validatorGroup, a struct"`

Reply:
I am not quite sure whether it is because of my terrible db connection but we now get past this query creation line but then get a panic directly below it when executing the query.